src/search_algo.hpp

```diff
@@ -598,7 +598,13 @@ _setFrames(TBlastMatch                          & bm,
         bm.qFrameShift = (m.qryId % 3) + 1;
         if (m.qryId % 6 > 2)
             bm.qFrameShift = -bm.qFrameShift;
-    } else if constexpr (seqan::qHasRevComp(TLocalHolder::TGlobalHolder::blastProgram))
+    } else if constexpr (TLocalHolder::TGlobalHolder::c_redAlph == AlphabetEnum::DNA3BS)
+    {
+        bm.qFrameShift = 1;
+        if (m.qryId % 4 > 1)
+            bm.qFrameShift = -bm.qFrameShift;
+    } else if constexpr (seqan::qHasRevComp(TLocalHolder::TGlobalHolder::blastProgram) &&
+                         TLocalHolder::TGlobalHolder::c_redAlph != AlphabetEnum::DNA3BS)
```

Comment:
```suggestion
    } else if constexpr (seqan::qHasRevComp(TLocalHolder::TGlobalHolder::blastProgram))
```

Not required because if-else?

Reply:
True ..